presentation/src/main/java/dominando/android/presentation/BookDetailsViewModel.kt

```diff
@@ -15,38 +15,36 @@ import kotlinx.coroutines.flow.flowOn
 import kotlinx.coroutines.launch
 
 class BookDetailsViewModel(
+    private val bookId: String,
     private val viewBookDetailsUseCase: ViewBookDetailsUseCase
 ) : ViewModel() {
 
     private val state: MutableLiveData<ViewState<BookBinding>> = MutableLiveData()
+    val book = Transformations.map(state) { it.data }
 
-    fun getState(): LiveData<ViewState<BookBinding>> = state
+    init { loadBook() }
```

Comment:
Could we use a `@LifecycleEvent` instead?

Reply:
Using `@LifecycleEvent` instead of `init block`, it's necessary to check if the `book` variable is already loaded every time when the device is rotated to avoid a new repository request.
Using `init block` approach, the `book` variable will be loaded only once when the ViewModel is created. When the device is rotated ViewModel will return the retained value without extra checking.
What do you think?